src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Binder.cs

```diff
@@ -41,10 +41,9 @@ public static class Binder
                 binaryOperationFlags |= CSharpBinaryOperationFlags.LogicalOperation;
             }
 
-            return new CSharpBinaryOperationBinder(operation, isChecked, binaryOperationFlags, context, argumentInfo);
+            return new CSharpBinaryOperationBinder(operation, isChecked, binaryOperationFlags, context, argumentInfo).TryGetExisting();
```

Comment:
Possibly, but it is harder and may be done separately. We would still need to so most of the argument validation and such.

The primary goal is not to avoid allocation of the initial empty binder, which is a relatively small object and not a huge or frequent expense, but to reuse the old instance in order to establish L2 cache sharing.